packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -223,14 +223,21 @@ export default class OptionManager {
   }
 
   mergePresets(presets: Array<string | Object>, dirname: string) {
+    const presetsLoc = presets.map(val =>
+      (typeof val === "string")
+        ? resolve(`babel-preset-${val}`, dirname) || resolve(val, dirname)
+        : val);
+
+    if (presetsLoc.length == 0) {
+      throw new Error("Couldn't find presets at all");
+    }
+
     for (let val of presets) {
       if (typeof val === "string") {
         let presetLoc = resolve(`babel-preset-${val}`, dirname) || resolve(val, dirname);
         if (presetLoc) {
           let presetOpts = require(presetLoc);
           this.mergeOptions(presetOpts, presetLoc, presetLoc, path.dirname(presetLoc));
-        } else {
```

Comment:
right now if we have one valid preset and one invalid this error will be thrown anyway, its not cool. But we still need to throw an error if both presets are invalid, so above I iterated over presets and as far as map skip `null` items, then if both `val` are invalid then `presetsLoc` will be empty, so its the only case when we need to throw error